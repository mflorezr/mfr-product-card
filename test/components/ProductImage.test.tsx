import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('debe mostrar el componente correctamente con la imagen personalizada', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://www.google.com/url?sa=i&url=https%3A%2F%2Fes.wikipedia.org%2Fwiki%2FCaf%25C3%25A9&psig=AOvVaw1H7gi9c2ijcrSqiyT4bFwM&ust=1700777883981000&source=images&cd=vfe&opi=89978449&ved=0CBEQjRxqFwoTCOjvy77R2IIDFQAAAAAdAAAAABAE" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostarr el componente con la imagen del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
