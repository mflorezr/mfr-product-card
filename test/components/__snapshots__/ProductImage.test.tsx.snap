// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostarr el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product "
    className="productImg undefined"
    src="./coffee-mug.png"
  />
  hola
</div>
`;

exports[`ProductImage debe mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Product "
  className="productImg undefined"
  src="https://www.google.com/url?sa=i&url=https%3A%2F%2Fes.wikipedia.org%2Fwiki%2FCaf%25C3%25A9&psig=AOvVaw1H7gi9c2ijcrSqiyT4bFwM&ust=1700777883981000&source=images&cd=vfe&opi=89978449&ved=0CBEQjRxqFwoTCOjvy77R2IIDFQAAAAAdAAAAABAE"
/>
`;
